engines/osxnative/engine.go

```diff
@@ -0,0 +1,51 @@
+//go:generate go-composite-schema --unexported --required engine payload-schema.yml generated_payloadschema.go
+
+// Package osxnative implements the Mac OSX engine
+package osxnative
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type engine struct {
+	engines.EngineBase
+	log *logrus.Entry
+}
+
+type engineProvider struct {
+	extpoints.EngineProviderBase
+}
+
+func init() {
+	// Register the mac engine as an import side-effect
+	extpoints.EngineProviders.Register(engineProvider{}, "macosx")
+}
+
+func (e engineProvider) NewEngine(options extpoints.EngineOptions) (engines.Engine, error) {
+	return engine{log: options.Log}, nil
+}
+
+func (e engineProvider) ConfigSchema() runtime.CompositeSchema {
+	return runtime.NewEmptyCompositeSchema()
+}
+
+func (e engine) PayloadSchema() runtime.CompositeSchema {
+	return payloadSchema
+}
+
+func (e engine) NewSandboxBuilder(options engines.SandboxOptions) (engines.SandboxBuilder, error) {
+	taskPayload := options.Payload.(*payload)
```

Comment:
Yea, I have a little mixed feelings about it too and no strong opinion either way.

The one decision we need to make is if this should be a "panic" event, which it will be if a type conversion fails or if it's something that we should fail on but continue to keep the worker running.